examples/qtmegachatapi/chatWindow.cpp

```diff
@@ -1330,6 +1352,44 @@ void ChatWindow::onTransferFinish(::mega::MegaApi* , ::mega::MegaTransfer *trans
     }
 }
 
+void ChatWindow::onRequestFinish(::mega::MegaApi *, ::mega::MegaRequest *request, ::mega::MegaError *e)
+{
+    switch (request->getType())
+    {
+        case ::mega::MegaRequest::TYPE_GET_ATTR_USER:
+            if (request->getParamType() == ::mega::MegaApi::USER_ATTR_PUSH_SETTINGS)
+            {
+                int currentDND = -1;
+                ::mega::m_time_t now = ::mega::m_time(NULL);
+                const ::mega::MegaPushNotificationSettings *currentSettings = request->getMegaPushNotificationSettings();
+                if (!currentSettings)
+                {
+                    currentSettings = ::mega::MegaPushNotificationSettings::createInstance();
+                }
+                else
+                {
+                    currentDND = currentSettings->getChatDnd(mChatRoom->getChatId()) - now;
+                    currentDND = (currentDND > -1) ? currentDND : -1;
+                }
+
+                bool ok = false;
+                int newDND = QInputDialog::getInt(this, tr("Push notification restriction - DND"),
+                                                             tr("Enter number of seconds without notifications for this chatroom: "), currentDND, -1, 2147483647, 1, &ok);
+                if (ok && currentDND != newDND)
+                {
+                    now = ::mega::m_time(NULL);
+                    ::mega::MegaPushNotificationSettings *newSettings = currentSettings->copy();
+                    newSettings->setChatDnd(mChatRoom->getChatId(), (newDND > -1) ? now + newDND : -1);
```

Comment:
This function doesn't allow to set `-1`. Instead, `MegaChatPushNotificationSettings::enableChat()` must be used.
I'll fix it